pkg/notify/notify.go

```diff
@@ -3,166 +3,218 @@ package notify
-			return r.reopen(issue.Key, logger)
-		}
+		level.Info(r.logger).Log("msg", "issue was recently resolved, reopening", "key", issue.Key, "label", issueGroupLabel)
+		return r.reopen(issue.Key)
+	}
+
+	if len(data.Alerts.Firing()) == 0 {
+		level.Debug(r.logger).Log("msg", "no firing alert; nothing to do.", "label", issueGroupLabel)
+		return false, nil
+	}
+
+	level.Info(r.logger).Log("msg", "no recent matching issue found, creating new issue", "label", issueGroupLabel)
+
+	issueType, err := r.tmpl.Execute(r.conf.IssueType, data)
+	if err != nil {
+		return false, errors.Wrap(err, "render issue type")
+	}
+
+	issueDesc, err := r.tmpl.Execute(r.conf.Description, data)
+	if err != nil {
+		return false, errors.Wrap(err, "render issue description")
 	}
 
-	level.Info(logger).Log("msg", "no recent matching issue found, creating new issue", "label", issueLabel)
 	issue = &jira.Issue{
 		Fields: &jira.IssueFields{
 			Project:     jira.Project{Key: project},
-			Type:        jira.IssueType{Name: r.tmpl.Execute(r.conf.IssueType, data, logger)},
-			Description: r.tmpl.Execute(r.conf.Description, data, logger),
-			Summary:     r.tmpl.Execute(r.conf.Summary, data, logger),
-			Labels: []string{
-				issueLabel,
-			},
-			Unknowns: tcontainer.NewMarshalMap(),
+			Type:        jira.IssueType{Name: issueType},
+			Description: issueDesc,
+			Summary:     issueSummary,
+			Labels:      []string{issueGroupLabel},
+			Unknowns:    tcontainer.NewMarshalMap(),
 		},
 	}
 	if r.conf.Priority != "" {
-		issue.Fields.Priority = &jira.Priority{Name: r.tmpl.Execute(r.conf.Priority, data, logger)}
+		issuePrio, err := r.tmpl.Execute(r.conf.Priority, data)
+		if err != nil {
+			return false, errors.Wrap(err, "render issue priority")
+		}
+
+		issue.Fields.Priority = &jira.Priority{Name: issuePrio}
 	}
 
-	// Add Components
 	if len(r.conf.Components) > 0 {
 		issue.Fields.Components = make([]*jira.Component, 0, len(r.conf.Components))
 		for _, component := range r.conf.Components {
-			issue.Fields.Components = append(issue.Fields.Components, &jira.Component{Name: r.tmpl.Execute(component, data, logger)})
+			issueComp, err := r.tmpl.Execute(component, data)
+			if err != nil {
+				return false, errors.Wrap(err, "render issue component")
+			}
+
+			issue.Fields.Components = append(issue.Fields.Components, &jira.Component{Name: issueComp})
 		}
 	}
 
-	// Add Labels
 	if r.conf.AddGroupLabels {
 		for k, v := range data.GroupLabels {
 			issue.Fields.Labels = append(issue.Fields.Labels, fmt.Sprintf("%s=%q", k, v))
 		}
 	}
 
 	for key, value := range r.conf.Fields {
-		issue.Fields.Unknowns[key] = deepCopyWithTemplate(value, r.tmpl, data, logger)
+		issue.Fields.Unknowns[key], err = deepCopyWithTemplate(value, r.tmpl, data)
+		if err != nil {
+			return false, err
```

Comment:
Maybe we should use `continue` here instead so that even if there's an issue with a single value, we still process the remaining ones?

Reply:
Yup, let's consider it later: https://github.com/free/jiralert/issues/36